### PR TITLE
Siamese Network + Classifier for the ADNI Brain Dataset - Shusheng Cao 46050021

### PR DESCRIPTION
**Pull request by Solarisxcao**
Hi,

This PR contains a classifier based on Siamese network to classify Alzheimer’s disease (normal and AD) of the ADNI brain data set (task 7). A classfier use a pre-train subnet of the Siamese network to classify Alzheimer’s disease.
Here's an overview of the main files:
1. `modules.py`:
    - This file contains the source code for the model's components.
    - All components are implemented in the form of classes.

2. `dataset.py`:
    - Responsible for loading and preprocessing the data.
    - Contains a data loader for fetching and processing images from the ADNI dataset.

3. `train.py`:
    - Contains the source code for training and saving the model.
    - The model is imported from modules.py, and the data loader is from dataset.py.
    - Losses and metrics are plotted during the training process.

4. `predict.py`:
    - Contains the source code for validaing and training the model.
 
5. `README.MD`:
    - Contains the model documentation for the project.
    - Provides instructions on installation, how to run the project, code structure, and background of the project, among other information.

For further details, please refer to the `README`.

Thank you!
Shusheng Cao
46050021

### CODE REVIEW

**Comment by nathasha-naranpanawa** [comment]
**This is an initial inspection, no action is required at this point**

Difficulty: Hard

Readme: OK
- sufficient information on the algorithm and usage
- no training hyperparameter details
- some loss/accuracy plots are provided, but no discussion on the results
- no prediction results included

Commit messages: OK

Code: 
- design is ok
- zero comments

Functionality/Performance: 
- siamese network seem to overfit after 20 epochs, but no other results are provided to check performance
- **classifier is only trained for 4 epochs and has 100% accuracy at every epoch, doesn't look very right**

General comments:
- solves the problem apporpriately, **performance is questionable**
- data leakage could have occured

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation
Good spacing and comments
Header blocks missing -1
## Recognition Problem 
Solves problem but has weird performance -2
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage
Module present
Commenting
No Data leakage, 100% accuracy, likely leakage -1
Difficulty: Hard
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe acceptable/good
Model/technical explanation
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No Feedback required
Request Description good
